docs/nx-cloud/intro/nx-agents.md

```diff
@@ -1,53 +1,141 @@
+Instead of defining
 
-## Revolutionizing Task Splitting
+```
+--distributes-on="8 linux-medium-js"
+```
+
+...which runs all tasks always on the same amount of machines, you can also have Nx Cloud scale the number of agents based on the size of your PR.
+
+```yaml {% fileName=".nx/workflows/dynamic-changesets.yaml" %}
+distributes-on:
+  small-changeset: 1 linux-medium
+  medium-changeset: 6 linux-medium
+  large-changeset: 10 linux-medium
+```
+
+{% callout type="note" title="How is the size of the PR being determined?" %}
+To determine the size of the PR, Nx Cloud calculates the relationship between the number of [affected projects](/ci/features/affected) vs the total number of projects in the workspace. It then assigns it to one of the three categories: small, medium, or large. This is static right now but might be configurable once Nx Agents is publicly available.
+{% /callout %}
+
+You can then reference it in your CI pipeline configuration:
+
+```yaml {% fileName=".github/workflows/main.yaml" %}
+...
+jobs:
+  - job: main
+    displayName: Main Job
+    ...
+    steps:
+      - checkout: self
+        persistCredentials: true
+      - script: |
+          npx nx-cloud start-ci-run --distributes-on=".nx/workflows/dynamic-changesets.yaml" --stop-agents-after="e2e-wrapper"
+        displayName: Start CI run
+      - ...
+```
 
-Consider an e2e project utilizing Cypress or Playwright. To leverage [affected](/ci/features/affected), [caching](/ci/features/remote-cache) and [distribution](/ci/features/distribute-task-execution), traditionally, you would need to break down your project into smaller, more manageable units. However, this can lead to a tedious and less efficient developer experience.
+## Automatic Task Splitting
 
-Nx Agents, in conjunction with a forthcoming Nx feature, solve this challenge by being able to dividing tasks (like e2e) into smaller, more refined units (e.g., at the file/test level). This enhances distribution across dynamically provisioned Nx Agents and can drastically reduce CI time.
+Imagine you're working on an end-to-end (e2e) project using tools like Cypress or Playwright. Traditionally, to make the most of features like [affected](/ci/features/affected), [caching](/ci/features/remote-cache), and [distribution](/ci/features/distribute-task-execution), you'd need to divide your project into smaller parts. But this approach can often be cumbersome and less efficient for developers.
+
+Nx is on the verge of introducing a game-changing feature that enables dynamic target definitions for projects (more details to come). Paired with Nx Agents, this innovation allows you to distribute e2e tests at the file level across various agents. (Combine that with flaky task detection and automated retries: see next section.
```

Comment:
`(Combine that with flaky task detection and automated retries: see next section.` This looks like a note to yourself that you didn't come back to.  I think you could delete this sentence and the section would work fine.  Or you could elaborate on what you mean.